neqo-transport/src/connection.rs

```diff
@@ -415,26 +423,48 @@ impl Connection {
 
     /// Call in to process activity on the connection. Either new packets have
     /// arrived or a timeout has expired (or both).
-    pub fn process_input(&mut self, dgram: Datagram, now: Instant) {
-        let res = self.input(dgram, now);
-        self.absorb_error(now, res);
+    pub fn process_input(&mut self, dgram: Option<Datagram>, now: Instant) {
```

Comment:
I wonder if this would be better as an unconditional processing functions and whether we could have a process_timers() function for managing the idle timeout and loss detection timeout.

Then callers could let arrival of packets drive one function and the expiration of timers drive the other.

The other way to cut this would be to have timers drive the process_output() function instead.  That is consistent with timers causing actions, like sending an ACK or retransmissions of lost packets.  I might even prefer that.

Reply:
I like a separate timer callback. Shall we try out that approach?

Reply:
I'll trust your judgment, particularly if you are the one to try things out.